src/main/kotlin/com/autonomousapps/tasks/ComputeAdviceTask.kt

```diff
@@ -254,10 +262,79 @@ internal class DependencyAdviceBuilder(
 
   private fun computeDependencyAdvice(projectPath: String): Sequence<Advice> {
     val declarations = declarations.filterToSet { Configurations.isForRegularDependency(it.configurationName) }
-    return dependencyUsages.asSequence()
+
+    /*
+     * To support KMP properly, we need to do an initial pass at deps to clean up any KMP deps advice.
+     *
+     * The primary case we are trying to catch and avoid here is one where we "fix" canonical KMP deps by trying to replace
+     * them with added deps on their specific targets ("-jvm", "-android", etc). To solve this, we save off information
+     * about both in the below mappings and then cross-reference them in the second pass farther down.
+     *
+     * Finally, in the third pass, we do our usual processing of advice but with the cleaned KMP advice. It's
```

Comment:
Three passes 😬 I haven't looked at the code below yet, but do you have a sense for the performance costs here? Also, should we just skip this stuff if we're not in a Kotlin project?

Reply:
it's more like 2.1 :). The middle pass is just looking at the KMP deps we collected in the first pass

Reply:
To answer your question, I'm not super fussed about the performance cost in this case. Still linear complexity. The absolute worst case is O(2.5n) (i.e. 50% of deps are KMP) if I'm remembering my algos and complexity right

Reply:
Alrighty, I'm willing to accept that analysis.